### PR TITLE
Create istio production overrides

### PR DESCRIPTION
**Pull request by Demonsthere**
**Description**

Changes proposed in this pull request:

- Create a production profile for istio
- Add doc with production overview
- ...

**Related issue(s)**

- Implements #6646

### CODE REVIEW

File: docs/service-mesh/05-05-production-profile.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
+---
+title: Service Mesh production profile
+type: Configuration
+---
+
+By default, every Kyma deployment is installed with the Service Mesh provider Istio, using what is considered a development profile. In this case, this means that:
+  - Horizontal Pod Autoscaler (HPA) is disabled for all components
```
Comment (piotrmsc) [inline docs/service-mesh/05-05-production-profile.md]: that applies to local environment, cluster configuration has HPA enabled

File: docs/service-mesh/05-05-production-profile.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        installer: overrides
+        component: istio
+        kyma-project.io/installation: ""
+    data:
+      global.proxy.resources.requests.cpu: "300m"
+      global.proxy.resources.requests.memory: "128Mi"
+      global.proxy.resources.limits.cpu: "500m"
+      global.proxy.resources.limits.memory: "1024Mi"
+      
+      gateways.istio-ingressgateway.resources.requests.cpu: "100m"
```
Comment (piotrmsc) [inline docs/service-mesh/05-05-production-profile.md]: this is default cluster value

File: docs/service-mesh/05-05-production-profile.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+      gateways.istio-ingressgateway.resources.requests.memory: "128Mi" 
+      gateways.istio-ingressgateway.resources.limits.cpu: "2000m"
```
Comment (piotrmsc) [inline docs/service-mesh/05-05-production-profile.md]: default value

File: docs/service-mesh/05-05-production-profile.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+This configuration is not considered production-ready. To use the Kyma Service Mesh in a production environment, configure Istio to use the production profile.
+
+## The production profile
+
+The production profile introduces the following changes to the Istio Service mesh:
+   - Resource quotas for all Istio components are increased: 
+   - All Istio components have HPA enabled
+   - HPA for Istio ingress-gateway has been customized:
+     + Minimal number of replicas: `3`
```
Comment (piotrmsc) [inline docs/service-mesh/05-05-production-profile.md]: 5 is defaut

File: docs/service-mesh/05-05-production-profile.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+      global.proxy.resources.requests.cpu: "300m"
+      global.proxy.resources.requests.memory: "128Mi"
+      global.proxy.resources.limits.cpu: "500m"
+      global.proxy.resources.limits.memory: "1024Mi"
+      
+      gateways.istio-ingressgateway.autoscaleMin: "3" 
+      gateways.istio-ingressgateway.autoscaleMax: "10"
+    EOF
+    ```
+  2. Run the cluster [update procedure](/root/kyma/#installation-update-kyma).
```
Comment (kazydek) [inline docs/service-mesh/05-05-production-profile.md]:
```suggestion
  2. Run the [cluster update procedure](/root/kyma/#installation-update-kyma).
```

File: docs/service-mesh/05-05-production-profile.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+By default, every Kyma deployment is installed with the Service Mesh provider Istio, using what is considered a development profile. In this case, this means that:
+  - Horizontal Pod Autoscaler (HPA) is enabled for all components, with the default numbers for replicas
+  - All components have severely reduced resource quotas, which is an performance factor.
+
+This configuration is not considered production-ready. To use the Kyma Service Mesh in a production environment, configure Istio to use the production profile.
+
+## The production profile
+
+The production profile introduces the following changes to the Istio Service mesh:
```
Comment (kazydek) [inline docs/service-mesh/05-05-production-profile.md]:
```suggestion
The production profile introduces the following changes to the Istio Service Mesh:
```

File: docs/service-mesh/05-05-production-profile.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+   - Resource quotas for all Istio components are increased: 
+   - All Istio components have HPA enabled
+   - HPA for Istio ingress-gateway has been customized:
+     + Minimal number of replicas: `3`
+     + Maximal number of replicas: `10`
```
Comment (kazydek) [inline docs/service-mesh/05-05-production-profile.md]:
```suggestion
     + Maximum number of replicas: `10`
```

File: docs/service-mesh/05-05-production-profile.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+  - All components have severely reduced resource quotas, which is an performance factor.
+
+This configuration is not considered production-ready. To use the Kyma Service Mesh in a production environment, configure Istio to use the production profile.
+
+## The production profile
+
+The production profile introduces the following changes to the Istio Service mesh:
+   - Resource quotas for all Istio components are increased: 
+   - All Istio components have HPA enabled
+   - HPA for Istio ingress-gateway has been customized:
```
Comment (kazydek) [inline docs/service-mesh/05-05-production-profile.md]:
```suggestion
   - HPA for Istio Ingress Gateway has been customized:
```

File: docs/service-mesh/05-05-production-profile.md [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+By default, every Kyma deployment is installed with the Service Mesh provider Istio, using what is considered a development profile. In this case, this means that:
+  - Horizontal Pod Autoscaler (HPA) is enabled for all components, with the default numbers for replicas
+  - All components have severely reduced resource quotas, which is an performance factor.
+
+This configuration is not considered production-ready. To use the Kyma Service Mesh in a production environment, configure Istio to use the production profile.
+
+## The production profile
+
+The production profile introduces the following changes to the Istio Service mesh:
+   - Resource quotas for all Istio components are increased:
```
Comment (kazydek) [inline docs/service-mesh/05-05-production-profile.md]:
```suggestion
   - Resource quotas for all Istio components are increased.
```

File: docs/service-mesh/05-05-production-profile.md [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+  - Horizontal Pod Autoscaler (HPA) is enabled for all components, with the default numbers for replicas
+  - All components have severely reduced resource quotas, which is an performance factor.
+
+This configuration is not considered production-ready. To use the Kyma Service Mesh in a production environment, configure Istio to use the production profile.
+
+## The production profile
+
+The production profile introduces the following changes to the Istio Service mesh:
+   - Resource quotas for all Istio components are increased: 
+   - All Istio components have HPA enabled
```
Comment (kazydek) [inline docs/service-mesh/05-05-production-profile.md]:
```suggestion
   - All Istio components have HPA enabled.
```

File: docs/service-mesh/05-05-production-profile.md [hunk 11]
```diff
@@ -0,0 +1,102 @@
+---
+title: Service Mesh production profile
+type: Configuration
+---
+
+By default, every Kyma deployment is installed with the Service Mesh provider Istio, using what is considered a development profile. In this case, this means that:
+  - Horizontal Pod Autoscaler (HPA) is enabled for all components, with the default numbers for replicas
```
Comment (kazydek) [inline docs/service-mesh/05-05-production-profile.md]:
```suggestion
  - Horizontal Pod Autoscaler (HPA) is enabled for all components, with the default number for replicas.
```

File: docs/service-mesh/05-05-production-profile.md [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+
+This configuration is not considered production-ready. To use the Kyma Service Mesh in a production environment, configure Istio to use the production profile.
+
+## The production profile
+
+The production profile introduces the following changes to the Istio Service Mesh:
+   - Resource quotas for all Istio components are increased.
+   - All Istio components have HPA enabled.
+   - HPA for Istio Ingress Gateway has been customized:
+     + Minimal number of replicas: `3`
```
Comment (kazydek) [inline docs/service-mesh/05-05-production-profile.md]:
```suggestion
     + Minimum number of replicas: `3`
```

File: docs/service-mesh/05-05-production-profile.md [hunk 13]
```diff
@@ -0,0 +1,102 @@
+---
+title: Service Mesh production profile
+type: Configuration
+---
+
+By default, every Kyma deployment is installed with the Service Mesh provider Istio, using what is considered a development profile. In this case, this means that:
+  - Horizontal Pod Autoscaler (HPA) is enabled for all components, with the default number for replicas.
+  - All components have severely reduced resource quotas, which is an performance factor.
```
Comment (kazydek) [inline docs/service-mesh/05-05-production-profile.md]:
```suggestion
  - All components have severely reduced resource quotas, which is a performance factor.
```

**Pull request merged**